test/functional/treesitter/parser_spec.lua

```diff
@@ -169,23 +169,21 @@ void ui_refresh(void)
 
   it("supports caching queries", function()
     local long_query = query:rep(100)
-    local first_run = exec_lua ([[
-      local before = vim.loop.hrtime()
-      cquery = vim.treesitter.parse_query("c", ...)
-      local after = vim.loop.hrtime()
-      return after - before
+    local first = exec_lua ([[
+      return tostring(vim.treesitter.parse_query("c", ...))
     ]], long_query)
 
-    local subsequent_runs = exec_lua ([[
-      local before = vim.loop.hrtime()
-      for i=1,100,1 do
-        cquery = vim.treesitter.parse_query("c", ...)
-      end
-      local after = vim.loop.hrtime()
-      return after - before
+    local second = exec_lua ([[
+      return tostring(vim.treesitter.parse_query("c", ...))
+    ]], long_query)
+
+    exec_lua [[collectgarbage()]]
```

Comment:
with this the test should be renamed to `supports cache invalidation`

Reply:
I am not sure, what I am checking here is that this is actually a weak table.

Reply:
Won't exec_lua always give you a copy of that table? You probably could use tostring to format the table as it's address.

Reply:
Earlier in the test I am testing that it actually returns the same table twice